### PR TITLE
Improved Provider Unit Test (use Prophecy)

### PR DESCRIPTION
**Pull request by dantleech**
Refactored `PhpcrProviderTestCase` to use Prophecy.

### CODE REVIEW

**Comment by dbu** [comment]
:+1:

File: Tests/Unit/Provider/PhpcrMenuProviderTest.php [hunk 1]
```diff
@@ -12,135 +12,90 @@
...
-            ->will($this->returnValue($session))
-        ;
-
-        return $dm;
+        $this->manager = $this->prophesize(DocumentManagerInterface::class);
+        $this->registry = $this->prophesize(ManagerRegistry::class);
+        $this->document = $this->prophesize(NodeInterface::class);
+        $this->item = $this->prophesize(ItemInterface::class);
+        $this->nodeLoader = $this->prophesize(NodeLoader::class);
+        $this->session = $this->prophesize(SessionInterface::class);
```
Comment (Tobion) [inline Tests/Unit/Provider/PhpcrMenuProviderTest.php]: all those properties have not been defined

Comment (dantleech) [inline Tests/Unit/Provider/PhpcrMenuProviderTest.php]: oops. my bad habit.

**Comment by WouterJ** [comment]
This should have been done in 2.0 imo, merge conflicts are very likely now.

**Comment by dantleech** [comment]
So we should switch to FQNs instead of `::class`?

**Pull request merged**